vendor/k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/util/proxy/upgradeaware.go

```diff
@@ -284,6 +284,16 @@ func (h *UpgradeAwareHandler) tryUpgrade(w http.ResponseWriter, req *http.Reques
 		rawResponse = headerBytes
 	}
 
+	// If the backend did not upgrade the request, return an error to the client. If the response was
+	// an error, the error is forwarded directly after the connection is hijacked. Otherwise, just
+	// return a generic error here.
+	if rawResponseCode != http.StatusSwitchingProtocols && rawResponseCode < 400 {
```

Comment:
here I'm using `rawResponseCode` directly vs `backendHTTPResponse.StatusCode` in 4.x